### PR TITLE
Generalised `cmdutils.Filter`

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

- extract parts of `cmdutils.NodeGroupFilter` into `cmdutils.Filter` for future re-use
- add `NameString` method to `api.NodeGroup`

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [ ] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Manually tested

### CODE REVIEW

**Comment by errordeveloper** [comment]
I wonder if I should create tests for `cmdutils.Filter`, most functionality is already covered in [`pkg/ctl/cmdutils/nodegroup_filter_test.go`](https://github.com/weaveworks/eksctl/blob/9903b4267daf25a0087d04366a46614b911f2023/pkg/ctl/cmdutils/nodegroup_filter_test.go)...

File: pkg/ctl/cmdutils/filter.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+// AppendExcludeGlobs sets globs for exclusion rules
+func (f *Filter) AppendExcludeGlobs(globExprs ...string) error {
+	for _, expr := range globExprs {
+		compiledExpr, err := glob.Compile(expr)
+		if err != nil {
+			return errors.Wrapf(err, "parsing glob filter %q", expr)
+		}
+		f.excludeGlobs = append(f.excludeGlobs, compiledExpr)
+		f.rawExcludeGlobs = append(f.rawExcludeGlobs, expr)
+
```
Comment (marccarre) [inline pkg/ctl/cmdutils/filter.go]: Über nit: extra empty line?

File: pkg/ctl/cmdutils/filter.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	if f.hasExcludeRules() {
+		logger.Info("exclude rules: %s", f.describeExcludeRules())
+		if excluded.Len() == 0 {
+			logger.Info("no %ss were excluded by the filter", resource)
+		}
+	}
+	if excluded.Len() > 0 {
+		logMsg(excluded, "excluded")
+	}
+}
```
Comment (marccarre) [inline pkg/ctl/cmdutils/filter.go]: Über nit: elsewhere in this commit you added a new line, so I'm guessing you may want to add one here too in order to be consistent?

File: pkg/ctl/cmdutils/filter.go (hunk 1 above)
Comment (errordeveloper) [reply pkg/ctl/cmdutils/filter.go]: Thanks!

File: pkg/ctl/cmdutils/filter.go (hunk 2 above)
Comment (errordeveloper) [reply pkg/ctl/cmdutils/filter.go]: No :)

**Comment by errordeveloper** [comment]
Closed via f32303d.

**Pull request closed without merging**